### PR TITLE
ResizeList component

### PR DESCRIPTION
**Pull request by swese44**
ResizeList will render a list of items on a single line, not rendering any items which don't fit in the available width. An optional `actionItem` can be provided, which will always be rendered after the other visible items.

This will support the Facepile-ish sidebar components:
https://app.zeplin.io/project/5907d6451d2b06c6b36b2852/screen/59c582c900581ab06b104f2f

Note that the design was changed and mockups were not updated. Zeplin currently shows a 6px gutter between each list item, which does not match our 4px gutters but stretched the list content to fully fill the current sidebar width on Yammer.com. The updated Design requirements asked for standard gutter sizes with the last "action" item aligned left.

## Pull request checklist

* [x] Component `README.md` file is up-to-date.
* [x] Component is unit tested.

### CODE REVIEW

File: src/components/ResizeList/ResizeList.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * ResizeList renders a items on a single row, without rendering any items which don't fit in the available width.
+ * An optional "actionItem" can be provided, which will always be rendered after the other visible items.
+ */
+export default class ResizeList extends React.Component<ResizeListProps> {
+  public render() {
+    const data: ResizeListData = {
+      count: this.props.items.length,
+      cacheKey: this.props.items.length.toString(),
```
Comment (unindented) [inline src/components/ResizeList/ResizeList.tsx]: This won't create problems if someone swaps one item with a different one?

File: src/components/ResizeList/ResizeList.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { join } from '../../util/classNames';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+import { ResizeGroup } from 'office-ui-fabric-react/lib/ResizeGroup';
```
Comment (unindented) [inline src/components/ResizeList/ResizeList.tsx]: Ha, I didn't know this existed. Do they rely on viewport `resize` events, or do they include a `ResizeObserver` polyfill?

Comment (swese44) [reply src/components/ResizeList/ResizeList.tsx]: they're literally listening to a throttled window resize event
https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/ResizeGroup/ResizeGroup.base.tsx#L342

File: src/components/ResizeList/ResizeList.tsx (hunk 1 above)
Comment (swese44) [reply src/components/ResizeList/ResizeList.tsx]: the cache just prevents recalculation, not a re-render, so everything is fine as long as the items don't change size. it looks like that's why the Fabric component offers a remeasure() public method, i'll surface that and componentRef

Comment (unindented) [reply src/components/ResizeList/ResizeList.tsx]: Ah fair.

Comment (swese44) [reply src/components/ResizeList/ResizeList.tsx]: actually that was a bit of a rabbit hole, `remeasure` only triggers the same container remeasure that window.resize triggers. looks like Fabric's ResizeGroup doesn't expect your items to change dimensions. we don't currently have this use case, but if we do we'll need to update Fabric's component to handle it

**Pull request merged**